src/StatsModels.jl

```diff
@@ -2,33 +2,64 @@ __precompile__(true)
 
 module StatsModels
 
-using Compat
-using DataFrames
+using Tables
 using StatsBase
-using Compat.SparseArrays
-using Compat.LinearAlgebra
-using Compat: @debug, @warn
+using CategoricalArrays
+using DataStructures
 
-export @formula,
-       Formula,
-       ModelFrame,
-       ModelMatrix,
+using SparseArrays
+using LinearAlgebra
+using Missings
 
-       AbstractContrasts,
-       EffectsCoding,
-       DummyCoding,
-       HelmertCoding,
-       ContrastsCoding,
+using Tables: ColumnTable
 
-       coefnames,
-       dropterm,
-       setcontrasts!
+export
+    #re-export from StatsBase:
+    StatisticalModel,
+    RegressionModel,
+    
+    @formula,
+    ModelFrame,
+    ModelMatrix,
 
+    AbstractContrasts,
+    EffectsCoding,
+    DummyCoding,
+    HelmertCoding,
+    ContrastsCoding,
+
+    coefnames,
+    dropterm,
+    setcontrasts!,
+
+    AbstractTerm,
+    TermOrTerms,
+    ConstantTerm,
+    Term,
+    ContinuousTerm,
+    CategoricalTerm,
+    InteractionTerm,
+    FormulaTerm,
```

Comment:
Are `FormulaTerm` and `MatrixTerm` really `AbstractTerm` objects, or do they need to be? I find it confusing to think of a formula as a term, and matrix terms are usually more than one (they are more like a block of terms or a sub-formula). Given the number of `*Term` types, it would make things easier to understand if only individual terms where called `*Term`.